lib/rubygems/user_interaction.rb

```diff
@@ -679,6 +679,7 @@ def initialize
     begin
       reader = File.open('/dev/null', 'r')
       writer = File.open('/dev/null', 'w')
+    rescue Errno::EINVAL
     rescue Errno::ENOENT
```

Comment:
Hmm, TIL. Like I said, I'm not a Ruby person; it's just a prerequisite for Vagrant.